app/src/main/java/com/eatssu/android/presentation/widget/ui/WidgetSettingActivity.kt

```diff
@@ -0,0 +1,172 @@
+package com.eatssu.android.presentation.widget.ui
+
+import android.appwidget.AppWidgetManager
+import android.content.Context
+import android.content.Intent
+import android.os.Build
+import android.os.Bundle
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
+import androidx.annotation.RequiresApi
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.stringPreferencesKey
+import androidx.datastore.preferences.preferencesDataStore
+import androidx.glance.GlanceId
+import androidx.glance.appwidget.GlanceAppWidgetManager
+import androidx.lifecycle.lifecycleScope
+import com.eatssu.android.data.enums.Restaurant
+import com.eatssu.android.presentation.widget.MealWorker
+import com.eatssu.design_system.theme.EatssuTheme
+import com.google.gson.Gson
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import timber.log.Timber
+
+val Context.dataStore by preferencesDataStore(name = "widget_prefs")
+
+@AndroidEntryPoint
+class WidgetSettingActivity : ComponentActivity() {
+    @RequiresApi(Build.VERSION_CODES.O)
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContent {
+            EatssuTheme {
+
+                val restaurantOptions = Restaurant.getVariableRestaurants().map {
+                    it.displayName
+                } // 변동 식당만 불러옵니다. 하드코딩 x
+
+                var selectedRestaurant by rememberSaveable { mutableStateOf(restaurantOptions[0]) }
+
+                val appWidgetId = intent?.getIntExtra(
+                    AppWidgetManager.EXTRA_APPWIDGET_ID,
+                    AppWidgetManager.INVALID_APPWIDGET_ID
+                )
+                val glanceId: GlanceId? =
+                    if (appWidgetId != null && appWidgetId != AppWidgetManager.INVALID_APPWIDGET_ID) {
+                        runBlocking {
+                            GlanceAppWidgetManager(this@WidgetSettingActivity).getGlanceIdBy(
+                                appWidgetId
+                            )
+                        }
```

Comment:
[ Suggestion - P1 ]
setContent는 UI 스레드로 알고 있는데 runblocking 을 사용하면 ANR이 난다고 합니다
suspend 호출은 LaunchedEffect나 lifecycleScope.launch로 비동기 처리하는 게 안전하다고 하네요

```suggestion
                       var glanceId by remember { mutableStateOf<GlanceId?>(null) }

LaunchedEffect(appWidgetId) {
    if (appWidgetId != null && appWidgetId != AppWidgetManager.INVALID_APPWIDGET_ID) {
        glanceId = GlanceAppWidgetManager(context).getGlanceIdBy(appWidgetId)
    }
}

```